src/uct/ib/rc/base/rc_iface.h

```diff
@@ -332,6 +332,7 @@ uct_rc_iface_put_send_op(uct_rc_iface_send_op_t *op)
 {
     uct_rc_iface_t *iface = op->iface;
     ucs_assert(op->flags & UCT_RC_IFACE_SEND_OP_FLAG_IFACE);
+    op->flags = UCT_RC_IFACE_SEND_OP_FLAG_IFACE;
```

Comment:
I agree, I was thinking about that as well.
Probably indeed flag should be dropped by op consumer.
Probably here it makes sense to add an assertion checking that all flags are cleared by consumer before returning the object.